zio-http/jvm/src/main/scala/zio/http/netty/AsyncBodyReader.scala

```diff
@@ -121,12 +177,52 @@ private[netty] abstract class AsyncBodyReader extends SimpleChannelInboundHandle
             !isLast
         }
 
+      // Reschedule timeout for next chunk if not the last message
+      if (readMore && !isLast) {
+        timeoutMillis.foreach { timeoutMillis =>
+          val task = ctx
+            .channel()
+            .eventLoop()
+            .schedule(
+              new Runnable {
+                override def run(): Unit = {
+                  AsyncBodyReader.this.synchronized {
```

Comment:
This is a bit confusing (I've actually never seen this syntax before). Maybe just use `this.synchronized`?